go/test/endtoend/backup/xtrabackup/xtrabackup_test.go

```diff
@@ -41,6 +48,49 @@ func TestXtrabackWithZstdCompression(t *testing.T) {
 	backup.TestBackup(t, backup.XtraBackup, "tar", 0, cDetails, []string{"TestReplicaBackup"})
 }
 
+func TestDoNotDemoteNewlyPromotedPrimaryIfReparentingDuringBackup(t *testing.T) {
+	defer cluster.PanicHandler(t)
+	clusterInstance := utils.SetupRangeBasedCluster(context.Background(), t)
+	clusterInstance.VtTabletExtraArgs = []string{
+		"--backup_engine_implementation", "xtrabackup",
+		"--xtrabackup_stream_mode=xbstream",
+		"--xtrabackup_user=vt_dba",
+		"--xtrabackup_backup_flags",
+		"--password=VtDbaPass",
+	}
```

Comment:
This won't work because the cluster has already been setup. The vttablets are already running by the time this code runs. You should add a parameter to the function you're using and then put these configurations before the vttablets are launched.

Reply:
Ok with enough finagling, I finally got this test to work 🎉